src/backend/distributed/executor/tuple_destination.c

```diff
@@ -203,8 +213,20 @@ TupleDestDestReceiverReceive(TupleTableSlot *slot,
 	HeapTuple heapTuple = ExecFetchSlotTuple(slot);
 #endif
 
+	DistributedExecutionStats *executionStats = tupleDestReceiver->executionStats;
+	if (SubPlanLevel > 0 && executionStats != NULL)
+	{
+		uint64 heapTupleSizeByte = HeapTupleHeaderGetDatumLength(heapTuple);
+		executionStats->totalIntermediateResultSize += heapTupleSizeByte;
+		if (CheckIfSizeLimitIsExceeded(executionStats))
```

Comment:
This does not seem like the right place. The underlying TupleDestination might not be storing intermediate results.

Can we do this in RemoteFileDestReceiverReceive?

Reply:
>  This does not seem like the right place. The underlying TupleDestination might not be storing intermediate results.

Hmm, I wanted to  imitate the adaptive executor. Wouldn't `SubPlanLevel > 0`  ensure we are  storing intermediate results?

> Can we do this in RemoteFileDestReceiverReceive?

I think the advantage of what we  do right now is the error  is thrown  as soon as we pass the limit while fetching the results over network (or locally pulling the data). 

Wouldn't `RemoteFileDestReceiverReceive` happen after all the data is fetched?

Reply:
makes sense, `TupleStoreTupleDestPutTuple` would be a better place, we can then remove the checks from adaptive executor as well